cmd/realm/triggers.go

```diff
@@ -177,3 +205,264 @@ func triggersDeleteF(command *cobra.Command, args []string) error {
+	if viper.GetBool("to-curl") {
+		fmt.Println(`'triggers sync' does not support the --to-curl option. Install or update your triggers one by one with 'triggers install' or 'interface update'.`)
+		os.Exit(1)
+	}
+
+	triggerToInstall := []map[string]interface{}{}
+	triggerToUpdate := []map[string]interface{}{}
+
+	for _, f := range args {
+		triggerFile, err := os.ReadFile(f)
+		if err != nil {
+			return err
+		}
+
+		var astarteTrigger map[string]interface{}
+		if err = json.Unmarshal(triggerFile, &astarteTrigger); err != nil {
+			return err
+		}
+
+		if _, err := getTriggerDefinition(realm, astarteTrigger["name"].(string)); err != nil {
+			// The interface does not exist
+			triggerToInstall = append(triggerToInstall, astarteTrigger)
+		} else {
+			triggerToUpdate = append(triggerToUpdate, astarteTrigger)
+		}
+
+		if len(triggerToInstall) == 0 && len(triggerToUpdate) == 0 {
+			// All good in the hood
+			fmt.Println("Your realm is in sync with the provided triggers files")
+			return nil
+		}
+
+		// Notify the user about what we're about to do
+	}
+
+	list := []string{}
+
+	for _, v := range triggerToInstall {
+		list = append(list, v["name"].(string))
+	}
+
+	list_existing := []string{}
+
+	for _, v := range triggerToUpdate {
+		list_existing = append(list_existing, v["name"].(string))
+	}
+
+	// Start syncing.
+
+	//install new triggers
+	if len(triggerToInstall) > 0 {
+
+		fmt.Printf("\n")
+		fmt.Printf("The following new triggers will be installed: %+q \n", list)
+		fmt.Printf("\n")
+
+		if ok, err := utils.AskForConfirmation("Do you want to continue?"); !ok || err != nil {
+			fmt.Printf("aborting")
+			return nil
+		}
+
+		for _, v := range triggerToInstall {
+			if err := installTrigger(realm, v); err != nil {
+				fmt.Fprintf(os.Stderr, "Could not install trigger %s: %s\n", v["name"].(string), err)
+			} else {
+				fmt.Printf("trigger %s installed successfully\n", v["name"].(string))
+			}
+		}
+
+		fmt.Printf("\n")
+
+	}
+
+	if len(triggerToUpdate) > 0 {
+
+		y, err := command.Flags().GetBool("force")
+		if err != nil {
+			return err
+		}
+
+		if y {
+			fmt.Printf("The following triggers already exists and WILL be DELETED and RECREATED: %+q \n", list_existing)
+			fmt.Printf("\n")
+			if ok, err := utils.AskForConfirmation("Do you want to continue?"); !ok || err != nil {
+				fmt.Printf("aborting")
+				return nil
+			}
+
+			for _, v := range triggerToUpdate {
+				if err := updateTrigger(realm, v["name"].(string), v); err != nil {
+					fmt.Fprintf(os.Stderr, "Could not update trigger %s: %s\n", v["name"].(string), err)
+				} else {
+					fmt.Printf("trigger %s updated successfully\n", v["name"].(string))
+				}
+			}
+			fmt.Printf("\n")
+			fmt.Printf("\n")
+
+		} else {
+
+			// Start syncing.
+			fmt.Printf("The following triggers already exists and WILL NOT be updated: %+q \n", list_existing)
+			fmt.Printf("\n")
+
+		}
+	}
+
+	return nil
+}
+
+func installTrigger(realm string, trigger map[string]interface{}) error {
```

Comment:
Can you use this also inside `triggersInstallF`? So that we don't duplicate code